c-api/dotnet/.editorconfig

```diff
@@ -0,0 +1,364 @@
+root = true
```

Comment:
Do you want to commit this?

Reply:
The `.editorconfig` file? Yes, that's how `dotnet format` is configured (it's also not bad for any editor supporting editorconfig)

Reply:
Oh ok!